Menus/TitleScreen.cpp

```diff
@@ -0,0 +1,608 @@
+				m_FadeAmount = static_cast<int>(LERP(0, 1.0F, 255.0F, 0, m_SectionProgress));
+				if (m_SectionElapsedTime >= m_SectionDuration) { SetTitleTransitionState((m_TitleTransitionState == TitleTransition::ScenarioFadeIn) ? TitleTransition::ScenarioMenu : TitleTransition::CampaignMenu); }
+				break;
+			case TitleTransition::FadeOut:
+				if (m_SectionSwitch) { SetSectionDurationAndResetSwitch(0.75F * g_SettingsMan.GetMenuTransitionDurationMultiplier()); }
+				g_AudioMan.SetTempMusicVolume(EaseIn(1.0F, 0, m_SectionProgress));
+				m_FadeAmount = static_cast<int>(EaseIn(0, 255, m_SectionProgress));
+				if (m_SectionElapsedTime >= m_SectionDuration) { SetTitleTransitionState(TitleTransition::TransitionEnd); }
+				break;
+			case TitleTransition::ScrollingFadeIn:
+				if (m_SectionSwitch) {
+					SetSectionDurationAndResetSwitch(0.75F * g_SettingsMan.GetMenuTransitionDurationMultiplier());
+					m_StationOrbitTimer.SetElapsedRealTimeS(m_StationOrbitTimerElapsedTime);
+					g_AudioMan.PlayMusic("Base.rte/Music/Hubnester/ccmenu.ogg", -1);
+				}
+				g_AudioMan.SetTempMusicVolume(EaseOut(0, 1.0F, m_SectionProgress));
+				m_ScrollOffset.SetY(EaseOut(250, 0, m_SectionProgress));
+				m_GameLogo.SetPos(Vector(static_cast<float>(g_FrameMan.GetResX() / 2), EaseOut(m_GameLogoPlanetViewOffsetY, m_GameLogoMainMenuOffsetY, m_SectionProgress)));
+				m_FadeAmount = static_cast<int>(EaseOut(255, 0, m_SectionProgress));
+				if (m_SectionElapsedTime >= m_SectionDuration) { SetTitleTransitionState(TitleTransition::MainMenu); }
+				break;
+			case TitleTransition::ScrollingFadeOut:
+			case TitleTransition::ScrollingFadeOutQuit:
+				if (m_SectionSwitch) { SetSectionDurationAndResetSwitch(0.75F * g_SettingsMan.GetMenuTransitionDurationMultiplier()); }
+				g_AudioMan.SetTempMusicVolume(EaseIn(1.0F, 0, m_SectionProgress));
+				m_ScrollOffset.SetY(EaseIn(0, 250, m_SectionProgress));
+				m_GameLogo.SetPos(Vector(static_cast<float>(g_FrameMan.GetResX() / 2), EaseIn(m_GameLogoMainMenuOffsetY, m_GameLogoPlanetViewOffsetY, m_SectionProgress)));
+				m_FadeAmount = static_cast<int>(EaseIn(0, 255, m_SectionProgress));
+				if (m_SectionElapsedTime >= m_SectionDuration) { SetTitleTransitionState((m_TitleTransitionState == TitleTransition::ScrollingFadeOutQuit) ? TitleTransition::TransitionEndQuit : TitleTransition::TransitionEnd); }
+				break;
+			case TitleTransition::TransitionEnd:
+				if (m_SectionSwitch) {
+					m_SectionSwitch = false;
+					m_TitleTransitionState = TitleTransition::TransitionEnd;
+				}
+				break;
+			default:
+				break;
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void TitleScreen::Draw() {
+		if (m_FinishedPlayingIntro) {
+			DrawTitleScreenScene();
+
+			// In credits have to draw the overlay before the game logo otherwise re-drawing it on top of an existing one causes glow effect to look wonky.
```

Comment:
I don't quite follow this comment, specifically `re-drawing it on top of an existing one causes...`
Existing what, game logo??

Reply:
`// In credits have to draw the overlay before the game logo otherwise re-drawing the game logo on top of an existing game logo causes glow effect to look wonky. Game logo.`

Reply:
Okay cool, wasn't sure if it meant redrawing the overlay or the game logo, though I guess the former doesn't make much sense. Anyway, probably tweak the wording in the comment to that if you haven't already, just for clarity.